src/MacVim/MMWindowController.m

```diff
@@ -160,6 +160,14 @@ - (id)initWithVimController:(MMVimController *)controller
 
     resizingDueToMove = NO;
 
+    afterWindowDidLoadQueueLock = [[NSLock alloc] init];
+    afterWindowDidLoadQueue = [[NSMutableArray alloc] init];
+    afterWindowDidLoadQueueTimer = [NSTimer scheduledTimerWithTimeInterval:0.25
```

Comment:
It seems like your theory is that the 'show dialog' messages comes in over the IPC channel after the window controller has been created but before the actual window has loaded (which seems like a reasonable theory).

Why a timer, though? It seems to me (without having an opportunity to sit at a Mac and test this yet) that the timer isn't needed; that runAfterWindowDidLoadUsingBlock could make a check to see if the window controller has already handled windowDidLoad and, if it has, execute the block immediately. Otherwise it pushes the block into the queue, which is executed during windowDidLoad (or I guess when we set "windowPresented").